doc/tutorials/chapter_7/00_sequential_logic.md

```diff
@@ -0,0 +1,291 @@
+# Content
+
+- [What is Sequential Logic?](#what-is-sequential-logic)
+- [Sequential Logic in ROHD](#sequential-logic-in-rohd)
+- [Shift Register](#shift-register)
+- [ROHD Simulator](#rohd-simulator)
+- [Unit Test in Sequential Logic](#unit-test-in-sequential-logic)
+- [Wave Dumper](#wave-dumper)
+- [Exercise](#exercise)
+
+## Learning Outcome
+
+In this chapter:
+
+- You will learn how to create a sequential logic that are equivalent to System Verilog `always_ff`. We will also review on what is `Simulator` and how to create unit test along with ROHD Simulator.
+
+## What is Sequential Logic?
+
+Sequential Logic is a circuit that consists of flip-flops as memory to generate output. In our previous chapter, we talked about combinational logic which input is only depend on the current state. But in sequential logic, past input will also be taken account before generate the current output.
+
+## Sequential Logic in ROHD
+
+In ROHD, a sequential module will consists of `Sequential()` class which map to system verilog `always_ff`. Just like `Combinationals()`, you can use `Conditionals()` inside `Sequential` block.
+
+Dart doesn't have a notion of certain signals being "clocks" vs "not clocks". You can use any signal as a clock input to sequential logic, and have as many clocks of as many frequencies as you want.
+
+## Shift Register
+
+So, let us look into how to implement a shift register in ROHD.
+
+A register is a digital circuit that use a group of flip-flops to store multiple bits of binary data (1 or 0). On the other hand, shift register is used to transfer the data in or out from register.
+
+Positive or Negative Edge of the clock signal is used to initiate the bit from moving around the register which make this a sequential logic circuit. In our example, we will be using positive edge triggered of the clock.
+
+Let start creating a shift register module and a main function to call on the shift register.
+
+```dart
+import 'package:rohd/rohd.dart';
+
+class ShiftRegister extends Module {
+  ShiftRegister();
+}
+
+void main() async {
+  final shiftReg = ShiftRegister();
+  await shiftReg.build();
+  print(shiftReg.generateSynth());
+}
+```
+
+Next, let define our inputs to the shift register. So, in our shift register we will need a reset pin `reset`, shift in pin `sin` and a clock `clk`. As for the output, there is one output pin shift out `sout`.
+
+Let register or add the inputs and output to our ShiftRegister module. Oh ya, we can also add the module name for our module as well. You can doing so by adding `super.name='shift_register'` to the constructor.
```

Comment:
> Oh ya, we can also add the module name for our module as well. You can doing so by adding `super.name='shift_register'` to the constructor.

Haha funny, but maybe not the right spot for this tidbit.  Maybe this should be in the modules section?

Reply:
Ya, let me pull this out for now.